src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,40 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden && this.health > 0) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        target.health = 0;
+        Animal.alive = Animal.alive.filter((animal) => animal !== target);
```

Comment:
The removal of dead animals from the `Animal.alive` array is done correctly using `filter`. This adheres to the checklist instruction to avoid nesting loops for performance reasons.